lib/utils.js

```diff
@@ -281,14 +281,21 @@ exports.createAssertionCheck = function (assertions, unexpectedGlobalAssertion,
             const isGlobal = isGlobalAssertion(node.callee);
             const assertion = isGlobal ? node.callee.name : node.callee.property.name;
 
-            context.report({
-                node: node,
-                messageId: isGlobalAssertion(node.callee) ? unexpectedGlobalAssertion : unexpectedLocalAssertion,
+            const reportErrorObject = {
+                node,
                 data: {
                     assertVar,
                     assertion
                 }
-            });
+            };
+            const errorMessageConfigForAssertion = errorMessageConfig[assertion];
+            if (errorMessageConfigForAssertion.unexpectedGlobalAssertionMessageId && errorMessageConfigForAssertion.unexpectedLocalAssertionMessageId) {
+                reportErrorObject.messageId = isGlobal ? errorMessageConfigForAssertion.unexpectedGlobalAssertionMessageId : errorMessageConfigForAssertion.unexpectedLocalAssertionMessageId;
+            } else {
+                reportErrorObject.message = isGlobal ? errorMessageConfigForAssertion.unexpectedGlobalAssertionMessage : errorMessageConfigForAssertion.unexpectedLocalAssertionMessage;
+            }
```

Comment:
All assumptions made here were properly checked in [this version](https://github.com/platinumazure/eslint-plugin-qunit/pull/80/commits/250023debde48ee61b1b54afcd6d56120d6c918c#diff-50e3aa130a4f97a42ee2cf111c7b1d9d) which would fail code coverage checks as the "default" paths are not tested. These can be re-enabled if the [rule recommended in this comment is eventually implemented](https://github.com/platinumazure/eslint-plugin-qunit/pull/80#issuecomment-633150211).